dagger/dag_creator/airflow/operators/spark_submit_operator.py

```diff
@@ -71,47 +73,12 @@ def execute(self, context):
         """
         See `execute` method from airflow.operators.bash_operator
         """
-        cmd = " ".join(
-            [
-                f"ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null",
-                f"hadoop@{self.emr_master} -tt",
-                self.spark_submit_cmd,
-            ]
+        response = self.ssm_client.send_command(
+            InstanceIds=[self.emr_master_instance_id],
+            DocumentName="AWS-RunShellScript",
+            Parameters={"commands": [self.spark_submit_cmd]},
         )
-
-        def pre_exec():
-            for sig in ("SIGPIPE", "SIGXFZ", "SIGXFSZ"):
-                if hasattr(signal, sig):
-                    signal.signal(getattr(signal, sig), signal.SIG_DFL)
-            os.setsid()
-
-        self.log.info(f"Running command: {cmd}")
-        self.sp = Popen(cmd.split(), stdout=PIPE, stderr=STDOUT, preexec_fn=pre_exec)
-        application_pattern = re.compile("(?<=proxy/)(.*)(?=/)")
-        pyspark_logs_only = False
-
-        while True:
-            line = self.sp.stdout.readline().decode("utf-8")
-            if line == "" and self.sp.poll() is not None:
-                break
-            if line:
-                if "tracking URL:" in line:
-                    self.application_id = application_pattern.findall(line)[0]
-                if "(state: FINISHED)" in line:
-                    pyspark_logs_only = False
-                if pyspark_logs_only:
-                    if "- pyspark - " not in line:
-                        continue
-                if "(state: RUNNING)" in line:
-                    pyspark_logs_only = True
-                self.log.info(line)
-
-        rc = self.sp.poll()
-
-        self.log.info(f"Command exited with return code {rc}")
-
-        if self.sp.returncode:
-            raise AirflowException("Bash command failed")
+        logging.INFO(response)
```

Comment:
Do we have all the logs in the response?

Reply:
Nope the logs is generated on the master node the response will just print the command id and other related fields